src/Processors/Transforms/Streaming/WatermarkStamper.cpp

```diff
@@ -315,9 +364,9 @@ void WatermarkStamper::processWatermark(Chunk & chunk)
     }
 }
 
-Int64 WatermarkStamper::calculateWatermark(Int64 event_ts) const
+Int64 WatermarkStamper::calculateWatermarkBasedOnWindowImpl(Int64 event_ts) const
```

Comment:
`calculateWatermarkImpl` since each subclass is already windowed based in the class name.